relay-cardinality/src/redis/quota.rs

```diff
@@ -112,16 +122,19 @@ impl QuotaScoping {
         let organization_id = self.organization_id.unwrap_or(0);
         let project_id = self.project_id.map(|p| p.value()).unwrap_or(0);
         let namespace = self.namespace.map(|ns| ns.as_str()).unwrap_or("");
-        let name = DisplayOptMinus(self.name.as_deref().map(fnv32));
+        let metric_type = DisplayOptMinus(self.metric_type);
+        let metric_name = DisplayOptMinus(self.metric_name.as_deref().map(fnv32));
 
         // Use a pre-allocated buffer instead of `format!()`, benchmarks have shown
         // this does have quite a big impact when cardinality limiting a high amount
         // of different metric names.
         let mut result = String::with_capacity(200);
         write!(
             &mut result,
-            "{KEY_PREFIX}:{KEY_VERSION}:scope-{{{organization_id}-{project_id}-{namespace}}}-{name}{slot}"
-        ).expect("formatting into a string never fails");
+            "{KEY_PREFIX}:{KEY_VERSION}:scope-{{{organization_id}-{project_id}-{namespace}}}-{metric_type}{metric_name}{slot}"
```

Comment:
Do we need to bump the KEY_VERSION for this?

Reply:
No, it's compatible, `type` and `name` are mutually exclusive and there is no overlap between name and type.

Reply:
Should it be an enum in `QuotaScoping` then?

Reply:
You caught me, yes it should be. But not only `name` and `type` should be in that enum it should be something like:

```rs
enum QuotaBlaScope {
    Organization {
        organization_id: OrganizationId,
    },
    Project {
        organization_id: OrganizationId,
        project_id: ProjectId,
    },
    Type {
        organization_id: OrganizationId,
        project_id: ProjectId,
        metric_type: MetricType,
    },
    Organization {
        organization_id: OrganizationId,
        project_id: ProjectId,
        metric_name: MetricName,
    },
}
```

And we should re-use this in the `CardinalityReport` and throughout the module.

I decided to just not do any enum for now since the code is very close together (QuotaScoping and its parent), but maybe we should.

Reply:
Up to you! Maybe it's something that can be handled in a follow-up PR?